scripts/object_hashing/create_object_md5.py

```diff
@@ -0,0 +1,121 @@
+import logging
+import subprocess
+
+logging.basicConfig(format="%(levelname)s: %(asctime)s : %(message)s", level=logging.INFO)
+
+def main():
+    description = """Generate md5 for object."""
+    parser = argparse.ArgumentParser(description=description)
+
+    parser.add_argument('-o', '--original_object', required=True, type=str, help='original object for md5 sum calculation')
+    parser.add_argument('-d', '--backup_dir', required=False, type=str, help='directory for creating backup of original object')
+    parser.add_argument('-r', '--requester_pays_project', required=False, default=None, type=str, help='directory for creating backup of original object')
+
+    args = parser.parse_args()
+    
+    original_bucket_name = args.original_object.split("/")[:-1][2] # fc-***
+    original_blob_name = "/".join(args.original_object.split("/")[3:]) # dir/filename.txt
+
+    # if the user provides a backup directory
+    if args.backup_dir:
+        create_backup_object(args.backup_dir, original_bucket_name, original_blob_name, args.requester_pays_project)
+    
+    # always create tmp object
+    create_tmp_object(original_bucket_name, original_blob_name, args.requester_pays_project)
+
+
+def create_md5_object(original_bucket_name, original_blob_name, tmp_blob_name, project_id=None):
+    """Replace original version with tmp object which now has md5."""
+    
+    logging.info("Starting replace of the original object with tmp object to generate md5.")
+    copy_object(original_bucket_name, tmp_blob_name, original_bucket_name, original_blob_name, project_id)
+
+    # get the md5 and write to file
+    final_obj_bytes, final_obj_md5 = get_file_size(original_bucket_name, original_blob_name, project_id)
+    logging.info(f"Final object md5: {final_obj_md5}")
+
+    with open("object_md5.txt", "w") as md5_file:
+        md5_file.write(final_obj_md5)
+
+
+def create_tmp_object(original_bucket_name, original_blob_name, project_id=None):
+    """Create tmp copy of original object."""
+
+    tmp_blob_name = f"{original_blob_name}.tmp"
+    logging.info(f"Starting creation of tmp copy to: gs://{original_bucket_name}/{tmp_blob_name}")
+    cmd = f"gsutil cp -D gs://{original_bucket_name}/{original_blob_name} gs://{original_bucket_name}/{tmp_blob_name}"
+    subprocess.run(cmd, shell=True)
+
+    compare_file_sizes(original_bucket_name, original_blob_name, original_bucket_name, tmp_blob_name, project_id)
+
+    # replace original with tmp 
+    create_md5_object(original_bucket_name, original_blob_name, tmp_blob_name, project_id)
+
+
+def create_backup_object(backup_directory, original_bucket_name, original_blob_name, project_id=None):
+    """Make copy of object in provided location."""
+
+    # TODO: check for slash at end
+    logging.info("Backup directory has been provided.")
+    backup_bucket_name = backup_directory.split("/")[:-1][2] # fc-
+    backup_blob_name = "/".join(backup_directory.split("/")[3:]) + "".join(original_blob_name.split("/")[-1:])
```

Comment:
You can use `os.path.join()`. This will make it so it doesn't matter if `/` is there or not